packages/protoc-gen-es/src/javascript.ts

```diff
@@ -43,7 +44,7 @@ export function generateJs(schema: Schema) {
 
 // prettier-ignore
 function generateEnum(schema: Schema, f: GeneratedFile, enumeration: DescEnum) {
-  const protoN = schema.runtime[enumeration.file.syntax];
+  const protoN = getNonEditionRuntime(schema, enumeration.file);
```

Comment:
The addition of `"editions"` to the `syntax` property will break code that only expects `"proto2" | "proto3"`, such as this expression. :disappointed:

Reply:
By "will break", does that mean this function would throw an exception in the face of "editions" syntax? If not, what exactly will happen here?

Reply:
In this specific code, nothing will happen. The `getNonEditionRuntime` will throw an error at runtime, but it never get's a chance to do so, because protoc-gen-es does not opt into edition support yet.

But the change I'm making here was necessary to avoid a compile-time error caused by `file.syntax` no longer being compatible as an index to `schema.runtime`.

Updated the PR description.